options/posix/generic/posix_stdio.cpp

```diff
@@ -18,9 +24,103 @@ int pclose(FILE *) {
 	__builtin_unreachable();
 }
 
-FILE *popen(const char*, const char *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+FILE *popen(const char *command, const char *typestr) {
+	bool is_write;
+	pid_t child;
+	FILE *ret = nullptr;
+
+	if (!mlibc::sys_fork || !mlibc::sys_close || !mlibc::sys_dup2 || !mlibc::sys_execve
+			|| !mlibc::sys_sigprocmask || !mlibc::sys_sigaction || !mlibc::sys_pipe) {
+		MLIBC_MISSING_SYSDEP();
+		errno = ENOSYS;
+		return nullptr;
+	}
+
+#if __MLIBC_POSIX_OPTION
+	pthread_testcancel();
```

Comment:
I don't see where ``popen()`` is specified as a cancellation point in POSIX.

Reply:
[This[(https://pubs.opengroup.org/onlinepubs/000095399/functions/xsh_chap02_09.html#tag_02_09_05_02) says that it _may_ be a cancellation point so I think that is correct.

Reply:
Ah I completely read over that, my bad. I'm still not sure we should include ``pthread_testcancel()`` however, since POSIX says it _can_ be a cancellation point to cover implementations that use blocking syscalls to implement it. In general library functions don't have to explicitely test for cancellations, since the sysdeps should already check for it appropriately.

E.g. nowhere in the ``fputc()`` implementation do we check using ``pthread_testcancel()``, but since it calls into ``read()``, it automatically becomes a cancellation point.

Reply:
Again, I did what `system()` did :^)

Reply:
``system()`` has the requirement that it _has_ to be a cancellation point, ``popen()`` doens't have this requirement. So imo it's not needed and can be removed.